Cron/DeleteBackups.php

```diff
@@ -0,0 +1,206 @@
+
+                    if($this->cronConfig->isLogsEnabled()) {
+                        $this->logger->debug(
+                            'CMS Content. Delete files by cron:',
+                            ['method' => CronConfig::PERIOD, 'files' => $deletedFiles]
+                        );
+                    }
+                }
+            } else if ($this->cronConfig->getMethodType() === CronConfig::OLDER_THAN) {
+                $filesOlderThan = $this->findFilesOlderThan($directoryIterator, $now);
+
+                if (count($filesOlderThan)) {
+                    $this->deleteFilesOlderThan($filesOlderThan);
+
+                    if($this->cronConfig->isLogsEnabled()) {
+                        $this->logger->debug(
+                            'CMS Content. Delete files by cron:',
+                            ['method' => CronConfig::OLDER_THAN, 'files' => $filesOlderThan]
+                        );
+                    }
+                }
+            }
+
+        }
+    }
+
+    /**
+     * @param $directoryIterator
+     * @param $now
+     * @return array
+     */
+    private function findAndSortFilesByPeriods($directoryIterator, $now): array
+    {
+        $files = $filesByWeeks = $filesByMonths = $filesByYears = [];
+
+        foreach (new \RecursiveIteratorIterator($directoryIterator) as $file) {
+            $filePath = $file->getPathname();
+            if (strpos($filePath, CronConfig::HISTORY_DIR)) {
+
+                if ($now - filemtime($filePath) >= CronConfig::WEEK && $now - filemtime($filePath) < CronConfig::MONTH) {
+                    $filesByWeeks[] = $filePath;
+                }
+
+                if ($now - filemtime($filePath) >= CronConfig::MONTH && $now - filemtime($filePath) < CronConfig::YEAR) {
+                    $filesByMonths[] = $filePath;
+                }
+
+                if ($now - filemtime($filePath) >= CronConfig::YEAR) {
+                    $filesByYears[] = $filePath;
+                }
+            }
+        }
+
+        if (count($filesByWeeks)) {
+            $files['week']['files'] = $filesByWeeks;
+            $files['week']['period'] = CronConfig::WEEK;
+        }
+
+        if (count($filesByMonths)) {
+            $files['month']['files'] = $filesByMonths;
+            $files['month']['period'] = CronConfig::MONTH;
+        }
+
+        if (count($filesByYears)) {
+            $files['year']['files'] = $filesByYears;
+            $files['year']['period'] = CronConfig::YEAR;
+        }
+
+        return $files;
+    }
+
+    /**
+     * @param $files
+     * @param $period
+     * @return array
+     */
+    private function deleteFilesByPeriod($files, $period): array
+    {
+        usort($files, function ($a, $b) { return filemtime($b) - filemtime($a); });
+
+        $previousFile = null;
+        $deletedFiles = [];
+
+        foreach ($files as $file) {
+            if (!$previousFile) $previousFile = $file;
+
+            if ($previousFile !== $file) {
```

Comment:
Seems with this rule the first $file in the array will not be processed, is this expected?